src/Native/CpuMath/tests/Performance/Perf.cs

```diff
@@ -0,0 +1,245 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+using Intrinsics;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Running;
+
+namespace PerformanceTests
+{
+    public class SsePerf
+    {
+        internal const int EXP_MAX = 127;
+        internal const int EXP_MIN = 0;
+        internal static float NextFloat(Random rand, int expRange)
+        {
+            double mantissa = (rand.NextDouble() * 2.0) - 1.0;
+            double exponent = Math.Pow(2.0, rand.Next(-expRange + 1, expRange + 1));
+            return (float)(mantissa * exponent);
+        }
+
+        private float[] src, dst, original, src1, src2;
+        private int[] idx;
+        private readonly int idxlen = 1000003;
+        private readonly int len = 1000003;
+        private readonly int expRange = EXP_MAX / 2;
+        private readonly int seed = 2;
+        private readonly float scale = 1.11f;
+
+        public SsePerf()
+        {
+            src = new float[len];
+            dst = new float[len];
+            src1 = new float[len];
+            src2 = new float[len];
+            original = new float[len];
+            Random rand = new Random(seed);
```

Comment:
This is such a good suggestion.  In our case, we want to generate and reproduce an input array of many elements, so a seed is introduced to set the entire array.  It will be useful to the tester to be able to test functions on different input arrays by setting different seeds.  

Not sure about the preferred style of writing performance tests.  Do we usually:
(1) write a hard-coded perf test with provided seeds that ensure everyone who runs the code to get the same, standardized result, or
(2) write a perf test that allows testers to customize the seeds and hence test with different input arrays?
Please let me know - thanks!

Reply:
I believe the general preference is to hard code the seed, but providing a mechanism to override the seed for additional (local/stress) testing is generally useful.